packages/shared/src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -15,19 +15,36 @@ export const RichTextEditor = ({
     placeholder,
     plugins,
     onBlur,
+    onTextChange,
+    onValueChanged,
     updateValueOnChange,
     showSerializedText,
+    shouldPreventPageLeave,
 }: RichTextEditorProps) => {
+    useEffect(() => {
+        const unloadHandler = (e: BeforeUnloadEvent) => {
+            e.preventDefault();
+            return (e.returnValue = 'Unprocessed changes');
+        };
+
+        if (shouldPreventPageLeave) {
+            window.addEventListener('beforeunload', unloadHandler);
+        }
+
+        return () => window.removeEventListener('beforeunload', unloadHandler);
+    }, [shouldPreventPageLeave]);
+
     if (isEditing) {
-        const saveText = (newValue: string) => newValue !== value && onBlur(newValue);
         return (
             <FondueRichTextEditor
                 id={id}
                 value={value}
                 border={false}
                 placeholder={placeholder}
-                onBlur={saveText}
```

Comment:
Is removing this function an issue? I imagine it was implemented for a specific reason